### DISCUSSION TITLE
Add an "Animation types" page

### DISCUSSION THREAD

**Discussion by yarusome** [mdn/mdn-community#367] [Content] [answered]
During the update in https://github.com/mdn/data/pull/660, I found that a lot of animation type data have become incorrect over the years since many properties now use one of the standard animation types defined in the [Web Animation](https://w3c.github.io/csswg-drafts/web-animations-1/#animating-properties) spec. Apart from correcting the remaining animation type data, I propose to:

  1. Add a new "Animation types" page.
  2. Make the "Animation type" key in "Formal definition" sections link to this new page.
  3. Remove the "Animatable CSS properties" [page](https://developer.mozilla.org/en-US/docs/Web/CSS/CSS_animated_properties), which is hard to maintain.

---

For the content on the "Animation types" page, I guess a reproduction from [5.2. Animating properties](https://w3c.github.io/csswg-drafts/web-animations-1/#animating-properties) does the job. However:

  1. The "(See prose)" type may be removed with an explaining paragraph added instead.
  2. The "by computed value" type refers to [3. Combining Values: Interpolation, Addition, and Accumulation](https://w3c.github.io/csswg-drafts/css-values-4/#combining-values) from the CSS Values and Units Module, so that information may be added to (1) the "CSS data types" [page](https://developer.mozilla.org/en-US/docs/Web/CSS/CSS_Types) or (2) the new "Animation types" page.

The advantage of 2(1) is that the pages are structured just like the specs, and the advantage of 2(2) is that readers don't have to click yet another link to see the details of combining values, which are currently only used by CSS transitions and animations anyway.

---

If the page outline is sorted out, I'm happy to make a tentative PR to materialize it.

---

Relates to https://github.com/mdn/content/issues/17768.

**Comment by teoli2003** [comment, marked as answer]
What about repurposing "Animatable CSS properties": instead of listing the useless list of animatable property (and getting rid of the macro), we list these predefined type (except prose).

**Reply by yarusome** [reply]
The repurposing seems good to me.

About the "combining values" part: after further investigation, I found that the pages for common CSS data types, e.g. [`<length>`](https://developer.mozilla.org/en-US/docs/Web/CSS/length#interpolation), already have the "Interpolation" section. Therefore new "Addtion" and "Accumulation" sections, if specs define them, can be added to the data type pages instead.
